geode-assembly/src/test/java/org/apache/geode/rest/internal/web/RestServersJUnitTest.java

```diff
@@ -15,43 +15,63 @@
 
 package org.apache.geode.rest.internal.web;
 
+import static org.apache.geode.distributed.ConfigurationProperties.HTTP_SERVICE_PORT;
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assume.assumeTrue;
 
-import org.apache.geode.distributed.internal.DistributionConfig;
-import org.apache.geode.test.dunit.rules.ServerStarterRule;
-import org.apache.geode.test.junit.categories.IntegrationTest;
-import org.apache.geode.test.junit.categories.RestAPITest;
 import org.apache.http.HttpResponse;
+import org.apache.http.HttpStatus;
 import org.json.JSONArray;
 import org.junit.BeforeClass;
 import org.junit.ClassRule;
 import org.junit.Test;
 import org.junit.experimental.categories.Category;
 
+import org.apache.geode.distributed.internal.DistributionConfig;
+import org.apache.geode.internal.AvailablePort;
+import org.apache.geode.test.dunit.rules.ServerStarterRule;
+import org.apache.geode.test.junit.categories.IntegrationTest;
+import org.apache.geode.test.junit.categories.RestAPITest;
+
 @Category({IntegrationTest.class, RestAPITest.class})
 public class RestServersJUnitTest {
 
+  private static boolean defaultPortWasAvailable;
+  private static int targetPort;
+
+  static {
+    defaultPortWasAvailable = AvailablePort
+        .isPortAvailable(DistributionConfig.DEFAULT_HTTP_SERVICE_PORT, AvailablePort.SOCKET);
+    targetPort = defaultPortWasAvailable ? DistributionConfig.DEFAULT_HTTP_SERVICE_PORT
+        : AvailablePort.getRandomAvailablePort(AvailablePort.SOCKET);
+  }
+
+
   @ClassRule
-  public static ServerStarterRule serverStarter =
-      new ServerStarterRule().withRestService(true).withAutoStart();
+  public static ServerStarterRule serverStarter = new ServerStarterRule().withRestService()
+      .withProperty(HTTP_SERVICE_PORT, targetPort + "").withAutoStart();
 
   private static GeodeRestClient restClient;
 
   @BeforeClass
   public static void before() throws Exception {
-    assertThat(serverStarter.getHttpPort()).isEqualTo(DistributionConfig.DEFAULT_HTTP_SERVICE_PORT);
     restClient = new GeodeRestClient("localhost", serverStarter.getHttpPort());
```

Comment:
what we can do is to put the assumeTrue clause here. If that passed, then start the server with default port.
You will need to get ride of withAutoStart() in the server declaration.

Reply:
And then after you started the server in the beforeCalss, you will need to assert that the port is indeed the default port.